packages/rum/src/domain/record/mutationObserver.spec.ts

```diff
@@ -431,6 +452,94 @@ describe('startMutationCollection', () => {
         ],
       })
     })
+
+    describe('for shadow DOM', () => {
+      it('should call shadowDomCreatedCallback when host is added', () => {
+        updateExperimentalFeatures(['record_shadow_dom'])
+
+        const serializedDocument = serializeDocumentWithDefaults()
+        const { mutationCallbackSpy, getLatestMutationPayload } = startMutationCollection()
+        const host = document.createElement('div')
+        const shadowRoot = host.attachShadow({ mode: 'open' })
+        shadowRoot.appendChild(document.createElement('span'))
+        sandbox.appendChild(host)
+        flushMutations()
+
+        expect(mutationCallbackSpy).toHaveBeenCalledTimes(1)
+        const { validate, expectNewNode, expectInitialNode } = createMutationPayloadValidator(serializedDocument)
+
+        const child = expectNewNode({ type: NodeType.Element, tagName: 'span' })
+        const shadowRootNode = expectNewNode({ type: NodeType.DocumentFragment, isShadowRoot: true }).withChildren(
+          child
+        )
+        const expectedHost = expectNewNode({ type: NodeType.Element, tagName: 'div' }).withChildren(shadowRootNode)
+        validate(getLatestMutationPayload(), {
+          adds: [
+            {
+              parent: expectInitialNode({ idAttribute: 'sandbox' }),
+              node: expectedHost,
+            },
+          ],
+        })
+        expect(addShadowRootSpy).toHaveBeenCalledOnceWith(shadowRoot)
+        expect(removeShadowRootSpy).not.toHaveBeenCalled()
+      })
+
+      it('should call shadowDomCreatedCallback when host is removed', () => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
      it('should call removeShadowRoot when host is removed', () => {
```